pkg/agent/datamodel/types_test.go

```diff
@@ -2039,15 +2039,18 @@ func TestKubernetesConfigGetOrderedKubeletConfigString(t *testing.T) {
 		expectedForPowershell string
 	}{
 		{
-			name:                  "zero value kubernetesConfig",
-			config:                &NodeBootstrappingConfiguration{},
+			name: "zero value kubernetesConfig",
```

Comment:
Should we keep the old unit test even if it is a negative test case that ContainerService is nil? We may also need another negative test case that Properties is nil?

Reply:
I checked these are several places assuming that `NodeBootstrappingConfiguration.ContainerService.Properties` is not nil, so I did not check the nil pointer in the newly introduced code.

Reply:
The old test case covers that ContainerService is nil but your PR removed this test case since you updated the existing unit test. You can keep it and add your new unit test

Reply:
Reverted the change on the old tests.